### PR TITLE
test(ogc): pin the QGIS >=3.44 CQL2 push-down conformance set

### PR DESCRIPTION
**Pull request by ishiland**
## Summary

Verified GeoLens's OGC API Features conformance advertisement against the exact QGIS >=3.44 requirement, before touching any code. Ground truth came from the merged fix itself: qgis/QGIS#62156 (merged 2025-06-05, first released in QGIS 3.44) patched `QgsOapifProvider::init()` (`src/providers/wfs/oapif/qgsoapifprovider.cpp`) to recognize the final-spec conformance class name `cql2/1.0/conf/basic-spatial-functions`, where it previously only recognized the deprecated draft name `cql2/{0.0,1.0}/conf/basic-spatial-operators`.

Reading that function directly shows two independent capability flags gate whether QGIS pushes a CQL2 filter server-side at all:

- `mServerSupportsFilterCql2Text` (base gate for ANY push-down) requires ALL of: `cql2/{0.0,1.0}/conf/basic-cql2`, `ogcapi-features-3/{0.0,1.0}/conf/filter`, `ogcapi-features-3/{0.0,1.0}/conf/features-filter`, `cql2/{0.0,1.0}/conf/cql2-text`.
- `mServerSupportsBasicSpatialFunctions` (gates spatial `S_INTERSECTS`/`BBOX()`/`POINT()` push-down specifically — what #62156 fixed) requires `cql2/1.0/conf/basic-spatial-functions` (or the deprecated `basic-spatial-operators` name).

Cross-checked against the CQL2 spec itself (`opengeospatial/ogcapi-features`, `cql2/standard/requirements/basic-spatial-functions/REQ_spatial-functions.adoc`): the `basic-spatial-functions` requirement class mandates only `S_INTERSECTS`, which matches what `backend/app/standards/ogc/filtering.py` implements (GeoLens actually supports the whole spatial-comparison family, a superset, and explicitly rejects `S_DWITHIN`/`S_BEYOND`, which belong to a different, unadvertised class).

**Result: GeoLens's current conformance list (`backend/app/standards/ogc/router.py`, from #1674/#1614) already advertises every one of the 5 required 1.0 URIs.** Both QGIS capability flags already evaluate `true`. No conformance-list change is honest to make, so this PR is test-only.

## Changes

- `backend/tests/test_ogc_discovery.py`: `test_conformance_pins_qgis_344_cql2_pushdown_classes` — pins the exact 5-URI set, cites qgis/QGIS#62156 and documents the two-flag detection logic in a comment.
- `backend/tests/test_ogc_features_filter.py`: `test_qgis_344_request_sequence_end_to_end` — replays the actual QGIS request sequence (landing → conformance → collection → queryables → items with a pushed-down `S_INTERSECTS(...)` filter, `filter-lang=cql2-text`) against the test app and asserts the conformance set plus correct filtered results.

Both tests guard against a future conformance-list edit silently breaking QGIS filter push-down — a regression here produces no error anywhere; QGIS just falls back to slow client-side filtering.

## Verification

```
cd backend && set -a && source ../.env.test && set +a && uv run pytest tests/test_ogc_features_filter.py tests/test_ogc_discovery.py -v   # 90 passed
uv run pytest tests/test_layering.py -q   # 47 passed
uv run ruff check .
uv run ruff format --check .
```

No product code changed, so `backend/openapi.json` and generated SDKs are unaffected.

https://claude.ai/code/session_01BXUUv3x71zWaLKEvoL4cTY

### CODE REVIEW

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

**Comment by ishiland** [comment]
@codex review

File: backend/tests/test_ogc_features_filter.py [hunk 1]
```diff
@@ -887,3 +888,183 @@ async def test_filter_still_rejected_without_value_types(
...
+    # trailing slash but never validates scheme or host. A misconfigured
+    # PUBLIC_API_URL of "https://" (no host) or "ftp://catalog.example"
+    # would make every generated link agree with that same broken origin,
+    # passing self-consistency while no real client could dereference any
+    # of it. Require a real HTTP(S) origin before trusting it as the
+    # baseline every other link gets checked against.
+    assert self_parsed.scheme in ("http", "https"), (
+        f"landing page self href {self_href!r} has a non-HTTP(S) scheme"
+    )
+    assert self_parsed.netloc, f"landing page self href {self_href!r} has no host"
```
[the remark this answers is not in the document]
Comment (ishiland) [reply backend/tests/test_ogc_features_filter.py:971]: Not fixing this one. This test replays QGIS's OGC API Features discovery sequence against the app under test and pins the exact conformance-class set QGIS >=3.44 requires (qgis/QGIS#62156) — its contract is that the server's advertised links are internally same-origin and followable, not that every conceivable PUBLIC_API_URL deployment misconfiguration (missing hostname, stray userinfo, malformed port, etc.) is individually rejected. That's config-validation scope, which belongs in `app/core/public_urls.py` (or a dedicated test of it), not in a QGIS-replay integration test. The origin class this thread and the two before it were probing is closed: the test already requires a real HTTP(S) scheme and a non-empty authority shared by every discovered link before trusting it, which is the invariant this test actually needs. Leaving this open as documentation for future reviewers rather than chasing further urlparse edge cases here.
